### PR TITLE
feat: send mail

### PR DESCRIPTION
**Pull request by yk-eukarya**

### CODE REVIEW

File: internal/usecase/interactor/emails/password_reset_html.tmpl [hunk 1]
```diff
@@ -0,0 +1,26 @@
+Content-Transfer-Encoding: quoted-printable
+Content-Type: text/html; charset=utf-8
+Mime-Version: 1.0
+
```
Comment (mimoham24) [inline internal/usecase/interactor/emails/password_reset_html.tmpl:4]: no need for those lines as content-type is generated by `SendMail()`
```suggestion
```

File: internal/usecase/interactor/emails/password_reset_text.tmpl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Content-Transfer-Encoding: quoted-printable
+Content-Type: text/plain; charset=utf-8
+Mime-Version: 1.0
```
Comment (mimoham24) [inline internal/usecase/interactor/emails/password_reset_text.tmpl:3]: same here
```suggestion

```

**Review by mimoham24 — CHANGES_REQUESTED** [review]

File: internal/usecase/interactor/user.go [hunk 3]
```diff
@@ -253,12 +266,42 @@ func (i *User) StartPasswordReset(ctx context.Context, email string) error {
...
 
-	u.SetPasswordReset(user.NewPasswordReset())
+	pr := user.NewPasswordReset()
+	u.SetPasswordReset(pr)
 
 	if err := i.userRepo.Save(ctx, u); err != nil {
 		return err
 	}
 
+	t1, err := textTmpl.New("passwordReset").Parse(passwordResetTextTMPL)
```
Comment (rot1024) [inline internal/usecase/interactor/user.go:276]: Make parsing templates in init() and put result in package variable. Another email template should be changed also.

```go
var emailTemplate *text.Template // Be careful of the variable name as it is a package variable

func init() {
	emailTemplate, err := textTmpl.New("passwordReset").Parse(passwordResetTextTMPL)
	if err != nil {
		log.Panicf("password reset email template parse error: %s", err)
	}
}
```

**Review by rot1024 — CHANGES_REQUESTED** [review]

File: internal/usecase/interactor/user.go [hunk 4]
```diff
@@ -29,9 +34,32 @@ type User struct {
...
+
+	passwordResetTextTMPL *textTmpl.Template
+	passwordResetHTMLTMPL *htmlTmpl.Template
+)
+
+func init() {
+	var err error
+	passwordResetTextTMPL, err = textTmpl.New("passwordReset").Parse(passwordResetTextTMPLStr)
+	if err != nil {
+		log.Panicf("password reset email template parse error: %s", err)
```
Comment (rot1024) [inline internal/usecase/interactor/user.go:55]:
```suggestion
		log.Panicf("password reset email template parse error: %s\n", err)
```

File: internal/usecase/interactor/user.go [hunk 5]
```diff
@@ -29,9 +34,32 @@ type User struct {
...
+	}
+	passwordResetHTMLTMPL, err = htmlTmpl.New("passwordReset").Parse(passwordResetHTMLTMPLStr)
+	if err != nil {
+		log.Panicf("password reset email template parse error: %s", err)
```
Comment (rot1024) [inline internal/usecase/interactor/user.go:59]:
```suggestion
		log.Panicf("password reset email template parse error: %s\n", err)
```

**Review by rot1024 — APPROVED** [review]

**Review by mimoham24 — APPROVED** [review]

**Pull request merged**